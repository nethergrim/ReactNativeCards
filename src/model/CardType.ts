
export enum CardType {
  Monster = 'Normal Monster',
  Spell = 'Spell Card',
  Trap = 'Trap Card',
}

export const detectCardType = (card) => {
  if (card.type.includes('Monster')) {
    return CardType.Monster;
  } if (card.type.includes('Spell')) {
    return CardType.Spell;
  } if (card.type.includes('Trap')) {
    return CardType.Trap;
  }

  return null;
};
