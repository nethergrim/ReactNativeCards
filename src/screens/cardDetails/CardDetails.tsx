import React from 'react';
import {
  FlatList, Image, Text, View,
} from 'react-native';

import { CardType, detectCardType } from '../../model/CardType';

import { styles } from './styles';

export const CardDetails = ({ route }) => {
  const { card } = route.params;
  const cardType = detectCardType(card);
  const smallImage = card.card_images[0].image_url;

  let backgroundStyle;
  switch (cardType) {
    case CardType.Monster:
      backgroundStyle = styles.monsterCard;
      break;
    case CardType.Trap:
      backgroundStyle = styles.trapCard;
      break;
    case CardType.Spell:
      backgroundStyle = styles.spellCard;
      break;
    default:
      break;
  }

  const renderSetItem = ({ item }) => {
    return (
      <View>
        <Text style={styles.attrText}>
          {'â€¢ '}
          {item.set_name}
          {' '}
          {item.set_rarity_code}

        </Text>
      </View>
    );
  };

  return (
    <View style={[styles.container, backgroundStyle]}>

      <View style={styles.horizontalContainer}>

        {card.level && (
          <Text style={styles.leftSideText}>
            Level
            {' '}
            {card.level}
          </Text>
        )}

        <Text style={styles.rightSideText}>
          {card.type}
          {' - '}
          {card.race}
        </Text>
      </View>

      <Image
        style={styles.logo}
        source={{
          uri: smallImage,
        }}
        resizeMode="contain"
      />

      <Text style={styles.titleText}>
        {card.name}
      </Text>

      {card.attribute && (
        <Text style={styles.attrText}>
          {card.attribute}
        </Text>
      )}

      <Text style={styles.descText}>
        {card.desc}
      </Text>

      <View style={styles.cardSpecsContainer}>
        <Text style={styles.attrText}>
          Def:
          {' '}
          {card.def}
        </Text>
        <Text style={styles.attrText}>{' / '}</Text>
        <Text style={styles.attrText}>
          Atk:
          {' '}
          {card.atk}
        </Text>
      </View>

      <Text style={styles.attrText}>
        Sets:
      </Text>
      <FlatList
        style={styles.setsList}
        data={card.card_sets}
        keyExtractor={(item) => item.set_code}
        renderItem={renderSetItem}
      />

    </View>
  );
};
