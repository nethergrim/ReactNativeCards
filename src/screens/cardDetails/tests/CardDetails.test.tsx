/* eslint-disable max-len */
import * as React from 'react';
import { render } from 'react-native-testing-library';

import { CardType } from '@model/CardType';

import { CardDetails } from '../CardDetails';

const testCard = {
  atk: 1150,
  attribute: 'LIGHT',
  card_images: [{ id: 58696829, image_url: 'https://storage.googleapis.com/ygoprodeck.com/pics/58696829.jpg', image_url_small: 'https://storage.googleapis.com/ygoprodeck.com/pics_small/58696829.jpg' }],
  card_prices: [{
    amazon_price: '0.79', cardmarket_price: '0.17', coolstuffinc_price: '0.25', ebay_price: '1.27', tcgplayer_price: '0.20',
  }],
  card_sets: [{
    set_code: 'LON-043', set_name: 'Labyrinth of Nightmare', set_price: '1.74', set_rarity: 'Common', set_rarity_code: '(C)',
  }, {
    set_code: 'LON-E043', set_name: 'Labyrinth of Nightmare', set_price: '1.81', set_rarity: 'Common', set_rarity_code: '(C)',
  }, {
    set_code: 'LON-EN043', set_name: 'Labyrinth of Nightmare', set_price: '2.45', set_rarity: 'Common', set_rarity_code: '(C)',
  }],
  def: 1000,
  desc: 'A mysterious priest created as a result of the latest advances in biotechnology.',
  id: 58696829,
  level: 3,
  name: 'Bio-Mage',
  race: 'Fairy',
  type: 'Normal Monster',
};

const props = {
  route: {
    params: {
      card: testCard,
    },
  },
};

describe('SearchListItem', () => {
  describe('render', () => {
    it('should render  SearchListItem', () => {
      expect(
        render(
          <CardDetails
            {...props}
          />,
        ).toJSON(),
      ).toMatchSnapshot();
    });
  });
});
