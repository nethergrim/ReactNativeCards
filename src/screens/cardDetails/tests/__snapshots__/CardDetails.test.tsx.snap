// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchListItem render should render  SearchListItem 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#ececec",
        "flex": 1,
        "padding": 12,
      },
      Object {
        "backgroundColor": "#C1855E67",
      },
    ]
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "flex": 1,
          "fontSize": 15,
          "fontWeight": "bold",
          "textShadowColor": "#000000aa",
          "textShadowOffset": Object {
            "height": 0.5,
            "width": 0.5,
          },
          "textShadowRadius": 1,
        }
      }
    >
      Level
       
      3
    </Text>
    <Text
      style={
        Object {
          "alignItems": "flex-end",
          "color": "#fff",
          "flex": 1,
          "fontSize": 16,
          "fontWeight": "bold",
          "justifyContent": "flex-end",
          "textAlign": "right",
          "textShadowColor": "#000000aa",
          "textShadowOffset": Object {
            "height": 0.5,
            "width": 0.5,
          },
          "textShadowRadius": 1,
        }
      }
    >
      Normal Monster
       - 
      Fairy
    </Text>
  </View>
  <Image
    resizeMode="contain"
    source={
      Object {
        "uri": "https://storage.googleapis.com/ygoprodeck.com/pics/58696829.jpg",
      }
    }
    style={
      Object {
        "alignSelf": "center",
        "height": 280,
        "marginBottom": 24,
        "marginTop": 24,
        "width": 160,
      }
    }
  />
  <Text
    style={
      Object {
        "alignSelf": "center",
        "color": "#fff",
        "fontSize": 22,
        "fontWeight": "bold",
        "marginBottom": 16,
        "textShadowColor": "#000000aa",
        "textShadowOffset": Object {
          "height": 0.5,
          "width": 0.5,
        },
        "textShadowRadius": 1,
      }
    }
  >
    Bio-Mage
  </Text>
  <Text
    style={
      Object {
        "color": "#fff",
        "fontSize": 15,
        "fontWeight": "bold",
        "marginBottom": 8,
        "textShadowColor": "#000000aa",
        "textShadowOffset": Object {
          "height": 0.5,
          "width": 0.5,
        },
        "textShadowRadius": 1,
      }
    }
  >
    LIGHT
  </Text>
  <Text
    style={
      Object {
        "color": "#fff",
        "fontSize": 18,
        "fontWeight": "500",
        "marginBottom": 8,
        "textShadowColor": "#000000aa",
        "textShadowOffset": Object {
          "height": 0.5,
          "width": 0.5,
        },
        "textShadowRadius": 1,
      }
    }
  >
    A mysterious priest created as a result of the latest advances in biotechnology.
  </Text>
  <View
    style={
      Object {
        "flexDirection": "row-reverse",
        "marginTop": 8,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontSize": 15,
          "fontWeight": "bold",
          "marginBottom": 8,
          "textShadowColor": "#000000aa",
          "textShadowOffset": Object {
            "height": 0.5,
            "width": 0.5,
          },
          "textShadowRadius": 1,
        }
      }
    >
      Def:
       
      1000
    </Text>
    <Text
      style={
        Object {
          "color": "#fff",
          "fontSize": 15,
          "fontWeight": "bold",
          "marginBottom": 8,
          "textShadowColor": "#000000aa",
          "textShadowOffset": Object {
            "height": 0.5,
            "width": 0.5,
          },
          "textShadowRadius": 1,
        }
      }
    >
       / 
    </Text>
    <Text
      style={
        Object {
          "color": "#fff",
          "fontSize": 15,
          "fontWeight": "bold",
          "marginBottom": 8,
          "textShadowColor": "#000000aa",
          "textShadowOffset": Object {
            "height": 0.5,
            "width": 0.5,
          },
          "textShadowRadius": 1,
        }
      }
    >
      Atk:
       
      1150
    </Text>
  </View>
  <Text
    style={
      Object {
        "color": "#fff",
        "fontSize": 15,
        "fontWeight": "bold",
        "marginBottom": 8,
        "textShadowColor": "#000000aa",
        "textShadowOffset": Object {
          "height": 0.5,
          "width": 0.5,
        },
        "textShadowRadius": 1,
      }
    }
  >
    Sets:
  </Text>
  <RCTScrollView
    data={
      Array [
        Object {
          "set_code": "LON-043",
          "set_name": "Labyrinth of Nightmare",
          "set_price": "1.74",
          "set_rarity": "Common",
          "set_rarity_code": "(C)",
        },
        Object {
          "set_code": "LON-E043",
          "set_name": "Labyrinth of Nightmare",
          "set_price": "1.81",
          "set_rarity": "Common",
          "set_rarity_code": "(C)",
        },
        Object {
          "set_code": "LON-EN043",
          "set_name": "Labyrinth of Nightmare",
          "set_price": "2.45",
          "set_rarity": "Common",
          "set_rarity_code": "(C)",
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    style={
      Object {
        "flex": 1,
      }
    }
    viewabilityConfigCallbackPairs={Array []}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View>
          <Text
            style={
              Object {
                "color": "#fff",
                "fontSize": 15,
                "fontWeight": "bold",
                "marginBottom": 8,
                "textShadowColor": "#000000aa",
                "textShadowOffset": Object {
                  "height": 0.5,
                  "width": 0.5,
                },
                "textShadowRadius": 1,
              }
            }
          >
            • 
            Labyrinth of Nightmare
             
            (C)
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View>
          <Text
            style={
              Object {
                "color": "#fff",
                "fontSize": 15,
                "fontWeight": "bold",
                "marginBottom": 8,
                "textShadowColor": "#000000aa",
                "textShadowOffset": Object {
                  "height": 0.5,
                  "width": 0.5,
                },
                "textShadowRadius": 1,
              }
            }
          >
            • 
            Labyrinth of Nightmare
             
            (C)
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View>
          <Text
            style={
              Object {
                "color": "#fff",
                "fontSize": 15,
                "fontWeight": "bold",
                "marginBottom": 8,
                "textShadowColor": "#000000aa",
                "textShadowOffset": Object {
                  "height": 0.5,
                  "width": 0.5,
                },
                "textShadowRadius": 1,
              }
            }
          >
            • 
            Labyrinth of Nightmare
             
            (C)
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
